File: Unique Characters (String)/is_unique.py
def is_unique(input):
  map = {}
  for i in range(0, len(input)):
    if map.has_key(input[i]):
      return False
    else:
      map[input[i]] = 1
  return True

print "The string, 'burger' is unique:"
print is_unique("burger")
print "The string, 'spinach' is unique:"
print is_unique("spinach")

File: Unique Characters (String)/is_unique.rb
def is_unique?(input)
  map = {}
  input = input.split("")
  input.each do |i|
    if map[i]
      return false
    else
      map[i] = 1
    end
  end
  true
end

puts "The string, 'burger' is unique:"
puts is_unique?("burger")
puts "The string, 'spinach' is unique:"
puts is_unique?("spinach")